recipes/python/build.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
```

Comment:
So let's think about this as setting up the standard for what most of our build scripts are gonna look like. I think there's opportunity here for removing lots of redundancy. I propose:

* we add our own `build_utils` package that gets built before anything else. Simple, pure python, no deps.
* We put reusable stuff like BuildConfig class in here
* having formatted cmake code embedded into py code isn't that easy to read. What if we had that code stored in a CMakeLists.txt.j2 file, and used Jinja2? I know that introduces another build-time dep, but damn if this wouldn't look a bit like ansible, and would help keep things pretty clean.

Reply:
I agree completely, however:
getting cmake and python built are two builds where some redundancy can be permitted because we're going to rely on them to build all the other packages, so we get a little bit chicken-egg-y. It depends on whether we want to allow the system-python-interpreter to be used as a bootstrapping dependency. What are your feelings there?

regarding templatizing cmakelists, I don't yet really see the value, but it's something we can move towards. For now, I'd like to target linux-build-ability, while we consider the best to provide multi-platform / multi-arch / multi-OS support, because we have some logistics issues there. For example, I think the best thing to do WRT to "platform-specific build flags" is to externalize the config from the build and load according to some dict-merge type of system, but what should that look like? I'm a little worried it'll end up looking like qt's 'mkspecs'